### PR TITLE
use a custom reverse in tests so no need to specify explicit namespaces

### LINKED ISSUE #8517: Implement custom resolve that resolves v3/v4 api-urls based on request
Our future testing-plan for v3/v4 APIs is that we intend to simply run V3 tests twice and test V4 with that, with custom overwrites if needed for V4 specific changes.

We will need to implement a custom resolve-helper that uses drf_resolve and force a namespace again for v3/v4 related urls.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #8517 

As `reverse_ns` defaults to `settings.REST_FRAMEWORK['DEFAULT_VERSION']` we can run an entire job with a different settings file that overrides it to `v3` or `v4` to run the API tests twice.

If we go down the route of doing it all the in same job then we need to split a number of classes into base and subclasses.  In that case `reverse_ns` could maybe be moved inside `TestClass`

### CODE REVIEW

**Comment by eviljeff** [comment]
Magic is https://github.com/mozilla/addons-server/pull/8557/files#diff-52c5a7aba86c1c829332505883eaf35bR1082

File: src/olympia/amo/tests/__init__.py [hunk 1]
```diff
@@ -1075,3 +1077,13 @@ def prefix_indexes(config):
 
     settings.CACHE_PREFIX = 'amo:{0}:'.format(prefix)
     settings.KEY_PREFIX = settings.CACHE_PREFIX
+
+
+def reverse_ns(viewname, api_version=None, args=None, kwargs=None, **extra):
```
Comment (diox) [inline src/olympia/amo/tests/__init__.py]: Let's add a docstring

Comment (EnTeQuAk) [inline src/olympia/amo/tests/__init__.py]: This needs some documentation. Especially mention that this generates a fake request on your behalf to resolve the URL.

File: src/olympia/amo/tests/__init__.py [hunk 2]
```diff
@@ -1075,3 +1077,13 @@ def prefix_indexes(config):
...
+    api_version = api_version or settings.REST_FRAMEWORK['DEFAULT_VERSION']
```
Comment (EnTeQuAk) [inline src/olympia/amo/tests/__init__.py]: It makes sense to use `rest_framework.settings:api_settings` here, e.g

```python
from from rest_framework.settings import api_settings
api_settings.DEFAULT_VERSION
```

which is the official interface to the `settings.REST_FRAMEWORK` object.

**Comment by EnTeQuAk** [comment]
> If we go down the route of doing it all the in same job then we need to split a number of classes into base and subclasses. In that case reverse_ns could maybe be moved inside TestClass

I'd rather go down the route of not duplicating entire jobs but only the `TestCase` classes and use `overwrite_settings` to overwrite the `DEFAULT_VERSION`. That'll make it much more explicit and easier to see what get's executed and simplifies debugging imho.

File: src/olympia/amo/tests/__init__.py (hunk 1 above)
Comment (eviljeff) [reply src/olympia/amo/tests/__init__.py]: https://github.com/mozilla/addons-server/pull/8557/commits/faf93f2f080e6d5432010a938a85fdcf8d9acb25

File: src/olympia/amo/tests/__init__.py (hunk 2 above)
Comment (eviljeff) [reply src/olympia/amo/tests/__init__.py]: https://github.com/mozilla/addons-server/pull/8557/commits/5b8de05f2f2d805a6d01410ed2f34456c3e72b38

**Pull request merged**